docs/linux/fs.md

```diff
@@ -6,127 +6,135 @@ sidebar_label: File Systems
-zoned block devices by allowing SSTables to be stored in a zone file similarly
-to a regular file system rather than as a range of sectors of the entire disk.
-The introduction of the higher level construct "one file is one zone" can help
-reducing the amount of changes needed in the application as well as introducing
-support for different application programming languages.
-
-The files representing zones are grouped by zone type, which are themselves
-represented by sub-directories. This file structure is built entirely using
-zone information provided by the device and so does not require any complex
-on-disk metadata structure.
+*zonefs* does not hide from the user the sequential write constraints of zoned
+block devices. In this, it is unlike a regular POSIX-compliant file system
+with native zoned-block device support (e.g. [*fsfs*](fs.md#f2fs)). Files
+that represent sequential write zones on the device must be written
+sequentially, starting from the end of the file (these are "append only"
+writes).
+
+*zonefs* is therefore more similar to a raw-block-device-access interface than
+it is to a full-featured POSIX file system. The goal of *zonefs* is to
+simplify the implementation of zoned block device support in applications, and
+it aims to do this by replacing raw block device file accesses with the richer
+regular-file API (which avoids relying on the possibly more obscure and
+developer-unfriendly direct block device file ioctls). One example of this
+approach is the implementation of LSM (log-structured merge) tree structures
+(such as used in RocksDB and LevelDB) on zoned block devices: SSTables are
+stored in a zone file in a way that is similar to the way a regular file
+system works rather than as a range of sectors of the entire disk. The
+introduction of the higher-level construct "one file is one zone" can reduce
+the number of changes needed in the application, and also introduces support
+for different application programming languages.
+
+The files that represent zones are grouped by zone type, and those zone types
+themselves are represented by sub-directories. This file structure is built
+entirely using zone information that is provided by the device and therefore
+does not require any complex on-disk metadata structure.
 
 ### On-Disk Metadata
 
 *zonefs* on-disk metadata is composed only of an immutable super block which
 persistently stores a magic number and optional feature flags and values. On
 mount, *zonefs* uses the block layer API function `blkdev_report_zones()` to
 obtain the device zone configuration and populates the mount point with a
-static file tree solely based on this information. File sizes come from the
-device zone type and write pointer position managed by the device itself.
+static file tree that is based solely on this information. File sizes come
+from the device zone type and the write-pointer position is managed by the
+device itself.
```

Comment:
The change changes the meaning... This probably should be like this:

...from the device zone type and the write-pointer position, which are both managed by the
device itself.

The point here is that zonefs only operates based on information from the device. zonefs does not have any metadata of its own.

Reply:
I have altered the sentence as you have suggested, and I have also put into the text your other statements about the point that zonefs just operates using information based on the device. I did this because I thought that they were helpful and on point.